apps/parsley/src/utils/filterLogs/index.ts

```diff
@@ -80,18 +80,27 @@ const filterLogs = (options: FilterLogsParams): ProcessedLogLines => {
       for (let idx = 0; idx < logLines.length; idx++) {
         const func = sectionData.functions[funcIndex];
         const isFuncStart = func && idx === func.range.start;
-        const isFuncOpen = sectionState[func?.functionID]?.isOpen ?? false;
+        const isFuncOpen =
+          (sectionState[func?.functionID]?.isOpen ||
+            func?.containsTopLevelCommand) ??
+          false;
```

Comment:
if there a reason we can't just automatically set the section to be open during processing if the function contains top level command?

Reply:
I rather handle this during processedLogLine construction (the render prop) so I can treat functions that contain top level commands the same as any other function when it comes to toggling all sections or any other open/close state logic we might add in the future.

Reply:
I refactored the code so the open state that's pushed to ProcessedLogLines is not related to `containsTopLevelCommand`

Reply:
I still think functions should be open if the command is visible and that the state should reflect that, but if the refactor will address that, then I'm fine with this for now

Reply:
Yup, that PR will address how top-level commands are represented in SectionState and SectionData by removing any function placeholders.